api.c

```diff
@@ -0,0 +1,1055 @@
+        {"keylayout", required_argument, NULL, 514},
+        {"noinput-text", required_argument, NULL, 515},
+        {"lock-text", required_argument, NULL, 516},
+        {"lockfailed-text", required_argument, NULL, 517},
+        {"greeter-text", required_argument, NULL, 518},
+        {"no-modkey-text", no_argument, NULL, 519},
+
+        // fonts
+        {"time-font", required_argument, NULL, 520},
+        {"date-font", required_argument, NULL, 521},
+        {"verif-font", required_argument, NULL, 522},
+        {"wrong-font", required_argument, NULL, 523},
+        {"layout-font", required_argument, NULL, 524},
+        {"greeter-font", required_argument, NULL, 525},
+
+        // text size
+        {"time-size", required_argument, NULL, 530},
+        {"date-size", required_argument, NULL, 531},
+        {"verif-size", required_argument, NULL, 532},
+        {"wrong-size", required_argument, NULL, 533},
+        {"layout-size", required_argument, NULL, 534},
+        {"modif-size", required_argument, NULL, 535},
+        {"greeter-size", required_argument, NULL, 536},
+
+        // text/indicator positioning
+        {"time-pos", required_argument, NULL, 540},
+        {"date-pos", required_argument, NULL, 541},
+        {"verif-pos", required_argument, NULL, 542},
+        {"wrong-pos", required_argument, NULL, 543},
+        {"layout-pos", required_argument, NULL, 544},
+        {"status-pos", required_argument, NULL, 545},
+        {"modif-pos", required_argument, NULL, 546},
+        {"ind-pos", required_argument, NULL, 547},
+        {"greeter-pos", required_argument, NULL, 548},
+
+        // text outline width
+        {"timeoutline-width", required_argument, NULL, 560},
+        {"dateoutline-width", required_argument, NULL, 561},
+        {"verifoutline-width", required_argument, NULL, 562},
+        {"wrongoutline-width", required_argument, NULL, 563},
+        {"modifieroutline-width", required_argument, NULL, 564},
+        {"layoutoutline-width", required_argument, NULL, 565},
+        {"greeteroutline-width", required_argument, NULL, 566},
+
+        // bar indicator stuff
+        {"bar-indicator", no_argument, NULL, 700},
+        {"bar-direction", required_argument, NULL, 701},
+        {"bar-orientation", required_argument, NULL, 703},
+        {"bar-step", required_argument, NULL, 704},
+        {"bar-max-height", required_argument, NULL, 705},
+        {"bar-base-width", required_argument, NULL, 706},
+        {"bar-color", required_argument, NULL, 707},
+        {"bar-periodic-step", required_argument, NULL, 708},
+        {"bar-pos", required_argument, NULL, 709},
+        {"bar-count", required_argument, NULL, 710},
+        {"bar-total-width", required_argument, NULL, 711},
+
+        {NULL, no_argument, NULL, 0}};
+
+    int o = 0;
+    int longoptind = 0;
+    char *optstring = "c:ui:tCFLMrs:kB:m";
+    char *arg = NULL;
+    int opt = 0;
+
+    char padded[9] = "ffffffff"; \
+
+#define parse_color(acolor)\
```

Comment:
I'd suggest turn these macros into functions.

Reply:
These are the same macros found in `main`. That would require to change `main` too (to avoid duplication).

Reply:
Ah ok. I know this is hard to do when we need to care about upstream. But I'd rather changing main than duplicating code. Duplicated code is always hard to maintain.